### DISCUSSION TITLE
[General] Measuring transactions per second (TPS) on zkSync Private Testnet

### DISCUSSION THREAD

**Discussion by ghost** [zkSync-Community-Hub/zksync-developers#708] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

Hey everyone!
I recently deployed a private testnet using zkSync and have been testing its performance. One of the key metrics I'm trying to evaluate is the Transactions Per Second (TPS) rate. I wanted to ask for suggestions on tools or methods to measure and visualize it accurately, we are using Grafana.

**Comment by alhenaai** [comment]
To measure and visualize the Transactions Per Second (TPS) on your zkSync private testnet using Grafana, you can follow these steps:

1. **Set Up Observability**: Ensure that your zkSync node's observability features are configured. This includes setting up metrics to be exposed, which you can then pull into Grafana.

2. **Configure Prometheus**: Use Prometheus to scrape the metrics from your zkSync node. You'll need to set up Prometheus to target your zkSync node's metrics endpoint.

3. **Grafana Dashboard**: In Grafana, create a dashboard to visualize the TPS. You can use the Prometheus data source to pull in the metrics. Specifically, you might want to track metrics related to transaction processing and block finalization.

4. **Custom Metrics**: If the default metrics do not include direct TPS measurements, you may need to create custom queries in Prometheus to calculate TPS based on the number of transactions in each block and the block time.

For detailed guidance on setting up observability for zkSync nodes, you can refer to the [ZKsync Node Observability documentation](https://docs.zksync.io/zksync-node/observability). This will provide you with information on configuring logs, metrics, and how to integrate with Prometheus and Grafana.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by popzxc** [comment, marked as answer]
Hey! We have public Grafana dashboards for the node, they include the TPS information. Check out this repository:
https://github.com/matter-labs/era-observability/

If you're using `zk_toolbox` to run you code, then it also has an option to configure the observability stack for you, check [its documentation](https://github.com/matter-labs/zksync-era/tree/main/zk_toolbox). Under the hood it still uses the `era-observability` repository, so it's fine to use whatever approach works for you.
